File: infer/tests/codetoanalyze/cpp/racerd/string_literal.cpp
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
extern int nondet_int();
extern char* nondet_string();

void loop() {
  int i = 0;
  char* s = nondet_string();

  while (nondet_int()) {
    s[i++] = "0123456789ABCDEF"[nondet_int() % 16];
  }
}

File: infer/tests/codetoanalyze/java/topl/strLiteral/StringLiteral.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
abstract class StringLiteral {
  abstract String source(String type);

  abstract void sink(String s);

  void fOk() {
    sink("ok");
  }

  void fBad() {
    sink("bad");
  }

  void gBad() {
    gLatent("bad");
  }

  void gLatent(String s) {
    sink(s);
  }

  void hOk() {
    sink(source("ok"));
  }

  void hBad() {
    sink(source("bad"));
  }
}
